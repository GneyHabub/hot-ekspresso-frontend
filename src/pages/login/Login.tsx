import {
  Box, Card, CardContent, CardHeader, IconButton, Input, InputAdornment, InputLabel, TextField,
} from '@mui/material';
import React from 'react';
import LoadingButton from '@mui/lab/LoadingButton';
import { useForm, Controller } from 'react-hook-form';
import { observer } from 'mobx-react';
import { AccountCircle, Visibility } from '@mui/icons-material';
import store from '../../store/store';

interface FormData {
  username: string;
  password: string;
}

const Login: React.FC = observer(() => {
  const { control, handleSubmit } = useForm<FormData>(
    {
      mode: 'onChange',
      defaultValues: {
        username: '',
        password: '',
      },
    },
  );
  const submit = (data: FormData) => {
    store.login(data.username, data.password);
  };
  return (
    <form
      style={{
        display: 'flex',
        justifyContent: 'center',
      }}
      onSubmit={handleSubmit(submit)}
    >
      <Card
        sx={{
          width: 500,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
        }}
      >
        <CardHeader title="Login" />
        <CardContent
          sx={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            gap: '20px',
          }}
        >
          <InputLabel htmlFor="standard-adornment-username">Username</InputLabel>
          <Controller
            name="username"
            control={control}
            render={({ field }) => (
              <Input
                {...field}
                id="standard-adornment-username"
                type="text"
                endAdornment={(
                  <InputAdornment position="end">
                    <IconButton
                      edge="end"
                    >
                      <AccountCircle />
                    </IconButton>
                  </InputAdornment>
        )}
              />
            )}
          />
          <InputLabel htmlFor="standard-adornment-password">Password</InputLabel>
          <Controller
            name="password"
            control={control}
            render={({ field }) => (
              <Input
                {...field}
                id="standard-adornment-password"
                type="password"
                endAdornment={(
                  <InputAdornment position="end">
                    <IconButton
                      aria-label="toggle password visibility"
                      edge="end"
                    >
                      <Visibility />
                    </IconButton>
                  </InputAdornment>
            )}
              />
            )}
          />
          <LoadingButton
            variant="contained"
            type="submit"
          >
            Log In
          </LoadingButton>
        </CardContent>
      </Card>
    </form>
  );
});

export default Login;
