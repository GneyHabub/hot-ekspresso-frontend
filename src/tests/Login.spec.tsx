import Login from '../pages/login/Login';
import {render} from '@testing-library/react'
import {BrowserRouter} from 'react-router-dom'

jest.mock('@auth0/auth0-react', () => ({
  useAuth0: () => ({
    loginWithRedirect: jest.fn
  })
}))


describe('Login', () => {
  it('matches the snapshot', () => {
    const wrapper = render(<BrowserRouter><Login/></BrowserRouter>);
    expect(wrapper.baseElement).toMatchSnapshot();
  })
})