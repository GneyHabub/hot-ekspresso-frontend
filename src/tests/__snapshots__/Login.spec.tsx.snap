// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1l4w6pd"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xg5b83-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <button
            class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            id="mui-1"
            tabindex="0"
            type="button"
          >
            Log In
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
